### PR TITLE
ARROW-10820: [Rust] [DataFusion] Complete TPC-H Benchmark Queries

### PR DESCRIPTION
**Pull request by seddonm1**
Changes:
- can now execute queries 3, 5 and 6 (in addition to 1, 12).
- add all tables.
- add all queries with query validation parameters provided.
- make basic modifications to queries 5 and 6 to allow supported syntax.
- queries 1 and 12 retain existing modifications.

this was an easy way to begin understanding some of the code base and get involved.

### CODE REVIEW

**Comment by seddonm1** [comment]
Sorry, put the wrong Jira ID when copying format 🤦

File: rust/benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -145,59 +145,849 @@ async fn benchmark(opt: BenchmarkOpt) -> Result<()> {
...
+            from
+                part,
+                supplier,
+                partsupp,
+                nation,
+                region
+            where
+                p_partkey = ps_partkey
+                and s_suppkey = ps_suppkey
+                and p_size = {SIZE}
```
Comment (Dandandan) [inline rust/benchmarks/src/bin/tpch.rs:221]: Why would we put the constants here in `format!` macro. Wouldn't it be better / easier to keep them as is, as they are directly coupled to the query anyway?

**Comment by Dandandan** [comment]
Awesome contribution, thanks @seddonm1  :) I added one comment

File: rust/benchmarks/src/bin/tpch.rs (hunk 1 above)
Comment (andygrove) [reply rust/benchmarks/src/bin/tpch.rs:221]: I think I agree. If we ever wanted to run these queries with different parameters we would want to use the tools provided by TPC to generate the queries to a text file. I'm not sure there is value in having them as parameters here.

**Review by andygrove — APPROVED** [review]
This is looking great. Thanks a lot @seddonm1

**Comment by andygrove** [comment]
It looks like formatting is off. See the Rust README for instructions on running cargo fmt.

File: rust/benchmarks/src/bin/tpch.rs (hunk 1 above)
Comment (seddonm1) [reply rust/benchmarks/src/bin/tpch.rs:221]: I am happy to change if you would like. 
My rationale:
- we can retain the information of which arguments are parameterised as per the TPC-H specification.
- we can retain 1:1 mapping of their query specification.
- the `format!` macro is compile time and should have very little performance impact

Comment (seddonm1) [reply rust/benchmarks/src/bin/tpch.rs]: I have updated PR without `format!` and with the TPC-H specified validation parameters hard coded.

**Review by Dandandan — APPROVED** [review]

**Pull request closed without merging**